### ISSUE TITLE
no commands are found for non-root users

### ISSUE THREAD

**Issue by cyleaf** [arkane-systems/genie#278]
I tried installing genie over a gentoo wsl from the tar.
I realize this is unsupported but I completely broke my system and I was hoping for a little hint if you happen to have an immediate idea on how to revert what breaking changes the tar did.. 

The effect is that no commands at all work for anyone non-root, even though $PATH is correct.
For example upon starting wsl, I get
-bash: awk: command not found
-bash: awk: command not found
and then it logs me in.

upon exiting, 
-bash: clear: command not found

essentially no commands work at all, which, sudo, etc.. including genie e.g. wsl genie -s gives:
/bin/bash: line 1: genie: command not found

So the system is basically not usable right now.
The root user has access to commands but I couldn't test because I then figured genie needs python's psutil but currently I can't emerge them because portage runs as a different user internally (i think) and since it can't find any commands, it says it can't find wget and fails.

**Comment by JeffBraxton** [comment]
Which directory did you untar it in?

**Comment by cyleaf** [comment]
> Which directory did you untar it in?

I untared to a tmp folder, then did sudo cp -rp ./* / when inside such that genie ends up in /usr/bin/genie

**Comment by JeffBraxton** [comment]
I think its supposed to be untarred as root in "/"

**Comment by cyleaf** [comment]
> I think its supposed to be untarred as root in "/"

no effects sadly. I did  ``tar xfv ./genie-systemd-2.3.tar.gz -C /`` as root. 


Like before, actually running genie gives this:
```
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/lib/genie/genie/__main__.py", line 14, in <module>
ModuleNotFoundError: No module named 'psutil'
```
Do you think it's possible this commands not found situation is due to the missing psutil module? If this is the cause, I could try to run a gentoo livecd from a usb, then mount the wsl2 vhdx, and somehow via chroot use the functional livecd system to install psutils on the wsl? It's a lot of black magic but I really can't think of anything.


Please let me know if you have any ideas!

**Comment by cerebrate** [comment]
genie does need the psutil module to run. As this is usually installed via a system package, you'd need to install that manually. It's _python3-psutil_ on Debian; don't know what it's called for Gentoo.

/usr should probably be mode 755, fwiw. You may want to check for similar issues on other directories mentioned in the tarball, such as /usr/bin, /usr/share/man, /usr/share/man/man8, etc., ideally versus a working Gentoo system. These permissions issues are the most likely cause for your problems, IMO. I'm going to make some changes in 2.4 to try and prevent a repeat.

**Comment by cyleaf** [comment]
I resolved it! 
Like you said, the problem with not finding the commands was due to incorrect permissions in the tar carrying over to the root system. I chmod-ed /usr, /usr/bin, /usr/share, and the rest back to the original 755, restarted, and things were back to normal.
Then I emerged the packages ``dev-python/psutil``, and I later found out I also need ``sys-apps/daemonize`` (for /usr/sbin/daemonize)
Afterwards, I did
```
sudo systemd-firstboot --prompt --setup-machine-id
sudo systemctl preset-all
```
and I believe also
```
sudo systemd-machine-id-setup
```
to finish the initial setup.

There was still a problem where systemd was in a degraded state. The failing service was ``user@0.service``, which I restarted once manually, and it was successful. To check I did a full restart, and then also ``systemctl list-units  --type=service  --state=running`` and everything was working without issues, but just in case as per genie's warning I switched to the multiuser target  ``sudo systemctl set-default multi-user.target``

And now I have a working systemd on my gentoo wsl! Thank you!
